packages/controllers/src/snaps/SnapController.ts

```diff
@@ -914,6 +907,21 @@ export class SnapController extends BaseController<
     );
   }
 
+  /**
+   * Safely revokes all permissions granted to a Snap
+   * @param snapId The snap ID
+   */
+  private revokeAllSnapPermissions(snapId: string) {
+    try {
+      this.messagingSystem.call(
+        'PermissionController:revokeAllPermissions',
+        snapId,
+      );
+    } catch {
+      // This may throw if the Snap has not been completely added yet
```

Comment:
Maybe catch and check if the error is related to the Snap not being completely added, and forward otherwise?

Reply:
Yeah, I'm not sure if there is other possibly errors that can be thrown from this. I guess possibly communication failure? Perhaps @rekmarks can elaborate.

I'm just trying to catch this: https://github.com/MetaMask/controllers/blob/main/src/permissions/PermissionController.ts#L868-L870

Reply:
Might be good practice anyway to catch the specific error, in case the function may throw other errors in the future.

Reply:
Updated in 5916bbf to circumvent the problem by checking if the snap has any permissions before revoking them.

As a completely separate matter, maybe we should have a way to "revoke" permissions that don't exist: https://github.com/MetaMask/controllers/issues/718